web/impact/impact/v1/views/office_hour_view.py

```diff
@@ -112,13 +115,21 @@ def handle_response(self, request):
             serializer = self.get_serializer(
                 instance, data=request.data, partial=True)
             save_operation = self.perform_update
-        else:
-            serializer = self.get_serializer(data=request.data)
-            save_operation = self.perform_create
         return self.perform_save(request, serializer, save_operation)
 
     def create(self, request, *args, **kwargs):
-        return self.handle_response(request)
+        data_sets = parse_date_specs(request.data)
+        serializers = [self.get_serializer(data=data) for data in data_sets]
+        invalid_serializers = [s for s in serializers if not s.is_valid()]
+        if invalid_serializers:
+            return handle_fail(invalid_serializers[0].errors)
+        for serializer in serializers:
+            self.perform_create(serializer)
+        office_hour = serializers[0].instance
+        if request.user != office_hour.mentor:
+            self.handle_send_mail(office_hour)
+
+        return handle_success(serializers[0].data)
```

Comment:
Non-blocker: Should we return every office hour in the block or just the first one?

Reply:
Interesting question. Depends what the front-end needs to have to do what it needs to do. 

I'm not super familiar with the requirements here, happy to make changes as needed.

Reply:
@PrinceDavis what do you think of this?

Reply:
return all sessions is the expectation, thanks @iankigen for highlighting

Reply:
@PrinceDavis I've made this change. Be aware that this means response.data['data'] is now a list